main/src/cgeo/geocaching/location/UTMPoint.java

```diff
@@ -0,0 +1,367 @@
+     * version of any valid letter passed in, 'A' .. 'Z'.
+     * 
+     * @throws NumberFormatException if zone letter is invalid.
+     */
+    private char checkZone(final char inZone) {
+        final char zone = Character.toUpperCase(inZone);
+
+        if (zone < 'A' || zone > 'Z') {
+            throw new NumberFormatException("Invalid UTMPoint zone letter: " + zone);
+        }
+
+        return zone;
+    }
+
+    /**
+     * Returns a string representation of the object.
+     * 
+     * @return String representation
+     */
+    public String toString() {
+        return String.format("%d%c E %d N %d", zoneNumber, zoneLetter, Math.round(easting), Math.round(northing));
+    }
+
+    /**
+     * Converts UTM coords to lat/long.
+     * <p>
+     * Equations from USGS Bulletin 1532 <br>
+     * East Longitudes are positive, West longitudes are negative. <br>
+     * North latitudes are positive, South latitudes are negative.
+     *
+     * @return Geopoint containing the lat/long value, or
+     *         <code>null</code> if conversion failed.
+     */
+    public Geopoint toLatLong() {
+        // check the ZoneNummber is valid
+        if (zoneNumber < 0 || zoneNumber > 60) {
+            return null;
+        }
+
+        // remove 500,000 meter offset for longitude
+        final double x = easting - FALSE_EASTING;
+        final double y = zoneLetter < 'N' ? northing - FALSE_NORTHING : northing;
+
+        // There are 60 zones with zone 1 being at West -180 to -174
+        final double longOrigin = (zoneNumber - 1) * 6 - 180 + 3; // +3 puts origin in middle of zone
+
+        final double m = y / K_0;
+        final double mu = m / (WGS_84_RADIUS * (1 - WGS_84_ECC_SQUARED / 4 - 3 * ECC_SQUARED_2 / 64 - 5 * ECC_SQUARED_3 / 256));
+
+        final double phi1Rad =
+                mu + (3 * E_1 / 2 - 27 * E_1 * E_1 * E_1 / 32) * Math.sin(2 * mu) + (21 * E_1 * E_1 / 16 - 55 * E_1 * E_1 * E_1 * E_1 / 32)
+                        * Math.sin(4 * mu) + (151 * E_1 * E_1 * E_1 / 96) * Math.sin(6 * mu);
+
+        final double n1 = WGS_84_RADIUS / Math.sqrt(1 - WGS_84_ECC_SQUARED * Math.sin(phi1Rad) * Math.sin(phi1Rad));
+        final double t1 = Math.tan(phi1Rad) * Math.tan(phi1Rad);
+        final double c1 = ECC_PRIME_SQUARED * Math.cos(phi1Rad) * Math.cos(phi1Rad);
+        final double r1 = WGS_84_RADIUS * (1 - WGS_84_ECC_SQUARED) / Math.pow(1 - WGS_84_ECC_SQUARED * Math.sin(phi1Rad) * Math.sin(phi1Rad), 1.5);
+        final double d = x / (n1 * K_0);
+
+        double lat =
+                phi1Rad
+                        - (n1 * Math.tan(phi1Rad) / r1)
+                        * (d * d / 2 - (5 + 3 * t1 + 10 * c1 - 4 * c1 * c1 - 9 * ECC_PRIME_SQUARED) * d * d * d * d / 24 + (61 + 90
+                        * t1 + 298 * c1 + 45 * t1 * t1 - 252 * ECC_PRIME_SQUARED - 3 * c1 * c1)
+                        * d * d * d * d * d * d / 720);
+        lat = radToDeg(lat);
```

Comment:
Why risk using mutable variables instead of using

``` java
   return new Geopoint(radToDeg(lat), radToDeg(lon));
```

?

(and maybe use `latRad` and `lonRad` as variable names)